def checkpoint_new_category_in_response():
    sv = h2o.upload_file(pyunit_utils.locate('smalldata/iris/setosa_versicolor.csv'))
    iris = h2o.upload_file(pyunit_utils.locate('smalldata/iris/iris.csv'))
    from h2o.estimators.gbm import H2OGradientBoostingEstimator
    m1 = H2OGradientBoostingEstimator(ntrees=100)
    m1.train(x=[0, 1, 2, 3], y=4, training_frame=sv)
    try:
        m2 = H2OGradientBoostingEstimator(ntrees=200, checkpoint=m1.model_id)
        m2.train(x=[0, 1, 2, 3], y=4, training_frame=iris)
        assert False, 'Expected continued model-building to fail with new categories introduced in response'
    except EnvironmentError:
        pass
